backends/dpdk/dpdkContext.cpp

```diff
@@ -411,6 +417,12 @@ Util::JsonArray *DpdkContextGenerator::addActions(const IR::P4Table *table,
                 for (auto param : *(attr.params)) {
                     if (param->type->is<IR::Type_Bits>()) {
                         param_width = param->type->width_bits();
+                        if (param_width % 8 != 0) {
+                            if (param_width < 32)
+                                param_width = 32;
+                            else
+                                param_width = 64;
+                        }
```

Comment:
is the param width more than 64 bits diagnosed earlier? Otherwise it is a silent truncation.

Reply:
Currently, we support upto 64-bits only. And this change is to keep the bitwidth in assembly file and context json in sync. I will add a BUG_CHECK perhaps. This may change in future, we will update the code then.